app/src/vuex/modules/connection.js

```diff
@@ -156,24 +157,11 @@ export default function ({ apollo }) {
       dispatch(`checkForPersistedSession`) // check for persisted session on that network
       // if the network is a pokadot network
       if (network.network_type === `polkadot`) {
-        actions.connectToPolkadotAPI({ commit }, network)
+        const polkadotAPI = await getPolkadotAPI(network)
```

Comment:
as described earlier you don't want to wait here until the api is connected. In ActionModal in a bad case, the api object would not be set yet.

Reply:
Done. At least what I think you meant